src/analysis/function_parameters.rs

```diff
@@ -311,3 +312,62 @@ fn get_length_type(
         array_length_type: array_length_type,
     }
 }
+
+fn detect_length<'a>(
+    env: &Env,
+    pos: usize,
+    par: &library::Parameter,
+    parameters: &'a [library::Parameter],
+) -> Option<&'a String> {
+    if !is_length(par) {
+        return None;
+    }
+
+    let array = parameters.get(pos - 1).and_then(
+        |p| if has_length(env, p.typ) {
+            Some(p)
+        } else {
+            None
+        },
+    );
+    array.map(|p| &p.name)
+}
+
+fn is_length(par: &library::Parameter) -> bool {
+    if par.direction != library::ParameterDirection::In {
+        return false;
+    }
+
+    if par.name.len() >= 3 && &par.name[0..3] == "len" {
```

Comment:
Usually it's ending with 'len', not beginning. 'objects_len', 'string_len', etc. Or with the full word 'length'.